src/Native/Unix/System.Native/pal_process.c

```diff
@@ -289,6 +309,13 @@ done:;
         CloseIfOpen(waitForChildToExecPipe[READ_END_OF_PIPE]);
     }
 
+#if !HAVE_PIPE2
+    if (haveProcessCreateLock)
+    {
+        pthread_mutex_unlock(&ProcessCreateLock);
+    }
+#endif
```

Comment:
Can we release the lock earlier?  Doing it here keeps the code tidy, but it also means we're holding the lock while, for example, waiting for the child process to write to the pipe, when it seems we could actually release the lock after the SystemNative_Pipe call.  If that's true, maybe keep this here, but also add:
```C++
#if !HAVE_PIPE2
    if (haveProcessCreateLock)
    {
        pthread_mutex_unlock(&ProcessCreateLock);
        haveProcessCreateLock = false;
    }
#endif

Reply:
I think I can move it above the wait but it must stay below the fork.